src/vm/stubgen.cpp

```diff
@@ -2770,7 +2797,11 @@ ILStubLinker::SetStubTargetArgType(
     TransformArgForJIT(pLoc);
 
     DWORD dwArgNum = m_nativeFnSigBuilder.NewArg(pLoc);
-    m_iTargetStackDelta--;
+
+    if (!m_fIsReverseStub)
```

Comment:
Why are we doing this? The fact that we are calling this for a P/Invoke regardless of inputs is a bit disconcerting.

Reply:
The call to SetStubTargetArgType adds an argument to the stub signature. When the stub is not a reverse stub, the stub signature is the signature that is called during the "dispatch" phase of the stub. So, we adjust the target stack delta to account for popping that argument off the IL stack.

Reply:
Lets comment that like we did above.